pandas/_libs/tslibs/util.pxd

```diff
@@ -228,5 +228,16 @@ cdef inline bint is_offset_object(object val):
     return getattr(val, '_typ', None) == "dateoffset"
 
 
-cdef inline bint _checknull(object val):
-    return val is None or (PyFloat_Check(val) and val != val)
+cdef inline bint is_nan(object val):
+    """
+    Check if val is a Not-A-Number float, including float('NaN') and np.nan.
+
+    Parameters
+    ----------
+    val : object
+
+    Returns
+    -------
+    is_nan : bool
+    """
+    return isinstance(val, float) and val != val
```

Comment:
should this be is_float_object (e.g. this excludes compex)

Reply:
This is functionally equivalent to the current implementation.

(The current version uses `PyFloatCheck(val)` instead of `isinstance(val, float)`, but cython converts the latter to the former)

Reply:
does this handle complex as well? I would rather use the same check as we do elsewhere, as this is confusing when to use one vs the other.

Reply:
> does this handle complex as well?

No.  This also only checks for python float, as opposed to is_float_object which also checks for numpy float (the examples I've checked with `isinstance(np.float64(2.3), float)` have come back `True`, not sure what the Falsey case is).

`_libs.missing.checknull` begins with the check you might have in mind:

```
if is_float_object(val) or is_complex_object(val):
    return val != val
[...]
```

ATM `_checknull` is used in exactly 19 places, at least one of which is in a float-only branch.  Based on a quick look it isn't obvious which usages (at least one) are specifically-for-float.  I'll take a look.

Reply:
After looking these over, I'm inclined to think it makes sense to use `is_float_object` but continue excluding complex.  At least two of the usages are floating-specific, and complex is sufficiently corner-case-y that I'd rather check for it explicitly-or-not-at-all.

It also looks like `checknull_with_nat` and `is_null_period` may be unnecessary, will look at that in an upcoming pass.